restcomm/restcomm.commons/src/main/java/org/restcomm/connect/commons/util/DigestAuthentication.java

```diff
@@ -97,6 +127,28 @@ private static void validate(final String user, final String realm, final String
             throw new NullPointerException("The uri parameter may not be null.");
         } else if (nonce == null) {
             throw new NullPointerException("The nonce parameter may not be null.");
+        } else if (algorithm == null) {
+            throw new NullPointerException("The algorithm parameter may not be null.");
         }
     }
+
+    /**
+     * @param username
+     * @param realm
+     * @param password
+     * @return
+     */
+    public static String HA1(String username, String realm, String password){
+        String algorithm = RestcommConfiguration.getInstance().getMain().getClientAlgorithm();
+        String ha1 = "";
+        ha1 = DigestAuthentication.H(username+":"+realm+":"+password, algorithm);
+        return ha1;
+    }
+
+    public static void main(String [] str){
+        String username="alice";
```

Comment:
should we turn this into unit test? or just remove it...

Reply:
we will remove it before merge

Reply:
will add a unit test as well